src/server.rs

```diff
@@ -3463,4 +3502,119 @@ sql"#;
+            },
+            work_done_progress_params: lsp::WorkDoneProgressParams {
+                work_done_token: None,
+            },
+            partial_result_params: lsp::PartialResultParams {
+                partial_result_token: None,
+            },
+        };
+
+        let result =
+            server.semantic_tokens_full(params).await.unwrap();
+        assert!(result.is_some());
+
+        let token_result = result.unwrap();
+        if let lsp::SemanticTokensResult::Tokens(tokens) =
+            token_result
+        {
+            let expected = lsp::SemanticTokens {
+                result_id: None,
+                data: vec![
+                    // package
+                    lsp::SemanticToken {
+                        delta_line: 1,
+                        delta_start: 1,
+                        length: 7,
+                        token_type: SEMANTIC_TOKEN_KEYWORD,
+                        token_modifiers_bitset: 0,
+                    },
+                    // "my-package"
+                    lsp::SemanticToken {
+                        delta_line: 1,
+                        delta_start: 9,
+                        length: 12,
+                        token_type: SEMANTIC_TOKEN_STRING,
+                        token_modifiers_bitset: 0,
+                    },
+                    // import
+                    lsp::SemanticToken {
+                        delta_line: 2,
+                        delta_start: 1,
+                        length: 6,
+                        token_type: SEMANTIC_TOKEN_KEYWORD,
+                        token_modifiers_bitset: 0,
+                    },
+                    // "csv"
+                    lsp::SemanticToken {
+                        delta_line: 2,
+                        delta_start: 8,
+                        length: 5,
+                        token_type: SEMANTIC_TOKEN_STRING,
+                        token_modifiers_bitset: 0,
+                    },
+                    // "my-bucket"
+                    lsp::SemanticToken {
+                        delta_line: 4,
+                        delta_start: 22,
+                        length: 11,
+                        token_type: SEMANTIC_TOKEN_STRING,
+                        token_modifiers_bitset: 0,
+                    },
+                    // 30m
+                    lsp::SemanticToken {
+                        delta_line: 5,
+                        delta_start: 21,
+                        length: 3,
+                        token_type: SEMANTIC_TOKEN_NUMBER,
+                        token_modifiers_bitset: 0,
+                    },
+                    // my.csv
+                    lsp::SemanticToken {
+                        delta_line: 7,
+                        delta_start: 16,
+                        length: 8,
+                        token_type: SEMANTIC_TOKEN_STRING,
+                        token_modifiers_bitset: 0,
+                    },
+                    // 0.9
```

Comment:
The running commentary of which tokens are what in the source is really helpful for keeping track.